### PR TITLE
allows Response::send_file() to correctly send file resources

### PR DESCRIPTION
**Pull request by leonar15**
This is a pretty straightforward fix that lets Koseven code generate files for download using tmpfile()

### CODE REVIEW

**Comment by WinterSilence** [comment]
Please, fix method description: set all types for $filename, add example.

**Comment by leonar15** [comment]
I've just added documentation and an exception handler for missing download name. Please take another look.

**Comment by WinterSilence** [comment]
@leonar15 * `@param   string|resource|bool  $filename   filename with path, file stream, or TRUE for the current response`

**Comment by WinterSilence** [comment]
@leonar15 also update tests.

**Comment by leonar15** [comment]
I'm trying to update the tests, but this function has a fundamental issue with testability since it uses "exit" to end script execution. This is probably why Response::send_file() has no current test coverage. Any suggestions?

**Comment by errotan** [comment]
Moving exit to a mockable class is an option.

https://stackoverflow.com/a/21578225

**Comment by leonar15** [comment]
@WinterSilence @errotan Are you suggesting I update the method so "exit" is called in a separate method? Kohana/seven has lots of places where exit is used that are equally untestable. It would be prudent to add a mockable exit() method to Core.

**Comment by WinterSilence** [comment]
@leonar15 Ok, no prombem - i dont see this test before. Wait other members for vote.

File: system/classes/Kohana/Response.php [hunk 1]
```diff
@@ -437,6 +443,17 @@ public function send_file($filename, $download = NULL, array $options = NULL)
 			// File data is no longer needed
 			unset($file_data);
 		}
+		else if ( ! empty($filename) && is_resource($filename))
```
[the remark this answers is not in the document]
Comment (errotan) [reply system/classes/Kohana/Response.php]: What is needed:

get_resource_type($res) == 'stream'

"stream" is for "dir handle", "file handle", "process handle" so another check:

$var = stream_get_meta_data($res);

$var['stream_type'] == 'STDIO' // on windows, haven't checked on linux

[the remark this answers is not in the document]
Comment (errotan) [reply system/classes/Kohana/Response.php]: or $var['seekable'] == TRUE

**Pull request merged**